File: Concesionario/src/baseDatos/Serializador.java
package baseDatos;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;
import java.io.PrintWriter;
import java.util.ArrayList;

import gestorAplicacion.activos.*;
import gestorAplicacion.personal.*;

public class Serializador {
	
	public static void serializarArrays(ArrayList<? extends Object> array, String name) {
		
		File archivo = new File("");
		
		try{
			File ruta = new File(archivo.getAbsolutePath()+"/src/baseDatos/temp/"+name+".txt");
			
			ObjectOutputStream escribiendo_fichero = new ObjectOutputStream(new FileOutputStream(ruta));
			
			escribiendo_fichero.writeObject(array);
			
			escribiendo_fichero.close();
			
		}
		catch (FileNotFoundException e) {
			System.out.print("FileNotFound");
		} catch (IOException e) {
			System.out.print("IOException");
		}
	}
	
	public static void serializarArrays() {
		Serializador.serializarArrays(InventarioAuto.getAutos(), "Autos");
		Serializador.serializarArrays(Cliente.getClientes(), "Clientes");
		Serializador.serializarArrays(Mecanico.getMecanicos(), "Mecanicos");
		Serializador.serializarArrays(Vendedor.getVendedores(), "Vendedores");
		Serializador.serializarArrays(InventarioArticulo.getArticulos(), "Articulos");
		Serializador.serializarArrays(InventarioArticulo.getRepuesto(), "Repuestos");
		Serializador.serializarArrays(Transaccion.getTransacciones(), "Transacciones");
		Serializador.serializarArrays(TransaccionVenta.getTransaccionesven(), "TransaccionesVentas");
		Serializador.serializarArrays(TransaccionVentaTaller.getTransaccionesven(), "TransaccionesVentaTaller");
		Serializador.serializarArrays(TransaccionTaller.getTransaccionestal(), "TransaccionesTaller");
		Serializador.serializarArrays(TransaccionModificacion.getTransaccionesmod(), "TransaccionesModificacion");
		Serializador.serializarArrays(TransaccionVenta.getAutosV(), "AutosV");
		Serializador.serializarArrays(TransaccionVenta.getMarcas(), "Marcas");
		Serializador.serializarArrays(TransaccionVenta.getVend(), "Vend");
		
	}
	
}

File: PYTHON/Concesionario/baseDatos/serializador.py
import pickle
import sys
import os
ruta_activos = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Activos'))
ruta_personal = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'gestorAplicacion', 'Personal'))
ruta_gestor = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
sys.path.append(ruta_activos)
sys.path.append(ruta_personal)
sys.path.append(ruta_gestor)
from gestorAplicacion.Personal.cliente import Cliente
from gestorAplicacion.Personal.mecanico import Mecanico
from gestorAplicacion.Personal.vendedor import Vendedor
from gestorAplicacion.Activos.inventario_articulo import Inventario_Articulo
from gestorAplicacion.Activos.InventarioAuto import InventarioAuto
from gestorAplicacion.Activos.transaccion import Transaccion
from gestorAplicacion.Activos.transaccionventa import TransaccionVenta
from gestorAplicacion.Activos.transaccionVentaTaller import TransaccionVentaTaller
from gestorAplicacion.Activos.TransaccionTaller import TransaccionTaller
from gestorAplicacion.Activos.TransaccionModificacion import TransaccionModificacion
from gestorAplicacion.Activos.articulo import Articulo
from gestorAplicacion.Activos.auto import Auto

class Serializador():
    
    def serializar_array(array, name):
        ruta = os.path.abspath(f"Concesionario/baseDatos/tmp/{name}.pkl")

        try:
            with open(ruta, "wb") as archivo:
                pickle.dump(array, archivo)
        except FileNotFoundError:
            print("FileNotFound")
        except IOError:
            print("IOError")

    def serializar_arrays():
        Serializador.serializar_array(Auto.get_autos(), "Autos")
        Serializador.serializar_array(Cliente.get_clientes(), "Clientes")
        Serializador.serializar_array(Mecanico.get_mecanicos(), "Mecanicos")
        Serializador.serializar_array(Vendedor.get_vendedores(), "Vendedores")
        Serializador.serializar_array(Articulo.get_articulos(), "Articulos")
        Serializador.serializar_array(Articulo.get_repuesto(), "Repuestos")
        Serializador.serializar_array(Transaccion.get_transacciones(), "Transacciones")
        Serializador.serializar_array(TransaccionVenta.get_transaccionesven(), "TransaccionesVentas")
        Serializador.serializar_array(TransaccionVentaTaller.get_transaccionesvental(), "TransaccionesVentaTaller")
        Serializador.serializar_array(TransaccionTaller.get_transaccionestal(), "TransaccionesTaller")
        Serializador.serializar_array(TransaccionModificacion.get_transaccionesmod(), "TransaccionesModificacion")
        Serializador.serializar_array(TransaccionVenta.get_autosV(), "AutosV")
        Serializador.serializar_array(TransaccionVenta.get_marcas(), "Marcas")
        Serializador.serializar_array(TransaccionVenta.get_vend(), "Vend")
